### DISCUSSION TITLE
Why Use Dependency Injection in WPF-UI Gallery Demos?

### DISCUSSION THREAD

**Discussion by FunJoo** [lepoco/wpfui#990] [Q&A] [answered]
Why does the WPF-UI Gallery demo utilize dependency injection? What are the benefits of employing the dependency injection pattern in WPF?

**Comment by pomianowski** [comment, marked as answer]
The dependency injection pattern provides easier unit testing, inversion of control and separation. 
It was used because the creators of the Gallery application found it to be a convenient solution, you do not have to use it in your application.
